semtypes/src/main/java/io/ballerina/types/definition/ObjectQualifiers.java

```diff
@@ -0,0 +1,69 @@
+/*
+ *  Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ *  WSO2 LLC. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package io.ballerina.types.definition;
+
+import io.ballerina.types.CellAtomicType;
+import io.ballerina.types.Env;
+import io.ballerina.types.PredefinedType;
+import io.ballerina.types.SemType;
+
+import java.util.List;
+
+import static io.ballerina.types.SemTypes.booleanConst;
+import static io.ballerina.types.SemTypes.stringConst;
+import static io.ballerina.types.SemTypes.union;
+import static io.ballerina.types.subtypedata.CellSubtype.cellContaining;
+
+public record ObjectQualifiers(boolean isolated, NetworkQualifier networkQualifier) {
+
+    public enum NetworkQualifier {
+        Client,
+        Service,
+        None;
+
+        private static final SemType CLIENT_TAG = stringConst("client");
+        private static final Field CLIENT = new Field("network", CLIENT_TAG, true, false);
+
+        private static final SemType SERVER_TAG = stringConst("service");
+        private static final Field SERVER = new Field("network", SERVER_TAG, true, false);
+
+        // Object can't be both client and server, which is enforced by the enum. We are using a union here so that
+        // if this is none it matches both
+        private static final Field NONE = new Field("network", union(CLIENT_TAG, SERVER_TAG), true, false);
+
+        private Field field() {
+            return switch (this) {
+                case Client -> CLIENT;
+                case Service -> SERVER;
```

Comment:
shall we use the same term as spec here? (SERVICE not SERVER)

Reply:
Fixed